### PR TITLE
MTDSA-14264: added TYS logic to AmendForeign endpoint, update unit te…

### PR DESCRIPTION
**Pull request by QHMRC**
…sts and iTest

### CODE REVIEW

File: app/v1/services/AmendForeignService.scala [hunk 1]
```diff
@@ -41,21 +41,29 @@ class AmendForeignService @Inject() (connector: AmendForeignConnector) extends D
       correlationId: String): Future[Either[ErrorWrapper, ResponseWrapper[Unit]]] = {
 
     val result = for {
-      desResponseWrapper <- EitherT(connector.amendForeign(request)).leftMap(mapDownstreamErrors(desErrorMap))
+      desResponseWrapper <- EitherT(connector.amendForeign(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
```
Comment (GazalNebHMRC) [inline app/v1/services/AmendForeignService.scala:44]: Please rename to downstreamResponseWrapper

File: test/v1/services/AmendForeignServiceSpec.scala [hunk 2]
```diff
@@ -93,7 +89,7 @@ class AmendForeignServiceSpec extends ServiceSpec {
             await(service.amendForeign(amendForeignRequest)) shouldBe Left(ErrorWrapper(correlationId, error))
```
Comment (GazalNebHMRC) [inline test/v1/services/AmendForeignServiceSpec.scala:89]: Please change desErrorCode to downstreamErrorCode

File: it/v1/endpoints/AmendForeignControllerISpec.scala [hunk 3]
```diff
@@ -425,13 +356,10 @@ class AmendForeignControllerISpec extends IntegrationBaseSpec {
 
       "ifs service error" when {
```
Comment (GazalNebHMRC) [inline it/v1/endpoints/AmendForeignControllerISpec.scala:357]: Change "ifs" to "downstream" in all the places in this test

File: it/v1/endpoints/AmendForeignControllerISpec.scala [hunk 4]
```diff
@@ -425,13 +356,10 @@ class AmendForeignControllerISpec extends IntegrationBaseSpec {
...
         def serviceErrorTest(ifsStatus: Int, ifsCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
-          s"ifs returns an $ifsCode error and status $ifsStatus" in new Test {
+          s"ifs returns an $ifsCode error and status $ifsStatus" in new TysIfsTest {
```
Comment (GazalNebHMRC) [inline it/v1/endpoints/AmendForeignControllerISpec.scala:359]: This should be NonTysTest

File: it/v1/endpoints/AmendForeignControllerISpec.scala [hunk 5]
```diff
@@ -448,7 +376,7 @@ class AmendForeignControllerISpec extends IntegrationBaseSpec {
              |}
```
Comment (GazalNebHMRC) [inline it/v1/endpoints/AmendForeignControllerISpec.scala:376]: This should be "downstream" too

File: it/v1/endpoints/AmendForeignControllerISpec.scala (hunk 4 above)
Comment (QHMRC) [reply it/v1/endpoints/AmendForeignControllerISpec.scala]: Is it better to have TysIfsTest here as we are also testing extraTysErrors?
It shouldnt matter too much as its mocking the endpoint calls

Comment (GazalNebHMRC) [reply it/v1/endpoints/AmendForeignControllerISpec.scala]: It does make sense to have TysIfsTest here. I suggested the change to have consistency with other endpoints. But, you are right it doesn't make much difference.

Comment (QHMRC) [reply it/v1/endpoints/AmendForeignControllerISpec.scala]: I been using TysIfsTest for my PRs. Im sure some are using the same as well

**Review by GazalNebHMRC — APPROVED** [review]

**Pull request merged**